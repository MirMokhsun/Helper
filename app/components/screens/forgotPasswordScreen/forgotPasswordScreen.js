import React, { Component } from 'react';
import { View, ScrollView, Text, KeyboardAvoidingView, StatusBar } from 'react-native';
import { styles } from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { CommonInput } from '../../commonComponents/commonInput/commonInput';
import { filterEmail } from '../../../api/regexFilters/regexFilters';
import { localization } from '../../../common/localization';
import ButtonSubmit from '../../commonComponents/buttonSubmit/buttonSubmit';
import Background from '../../commonComponents/svg/Background';
import Logo from '../../commonComponents/svg/logo';
import { setForgotPasswordEmail } from './redux/forgotPasswordDataAction';
import { showForgotPasswordSpiner } from '../../commonComponents/buttonSubmit/redux/spinersAction';
import { ButtonBack } from '../../commonComponents/buttonBack/buttonBack';
import { crmForgotPassword } from '../../../api/forgotPassword/forgotPasswordSingleton';
import Utils from '../../../api/utils/utils';

const { isIOS } = Utils;

export class ForgotPasswordScreen extends Component {
    static navigationOptions = () => {
        return {
            header: null,
        }
    };

    constructor(props) {
        super(props);
        this.state = {
            isForgotPasswordDisabled: true,
        }
    }

    UNSAFE_componentWillReceiveProps(nextProps) {
        const { isValid } = nextProps.forgotPasswordData.email;
        if (isValid && nextProps.isConnected) {
            this.setState({ isForgotPasswordDisabled: false });
        } else {
            this.setState({ isForgotPasswordDisabled: true });
        }
    }

    UNSAFE_componentWillUnmount() {
        const { setForgotPasswordEmail } = this.props;
        setForgotPasswordEmail({ isValid: false, value: null });
    }

    sendForgotPasswordData = () => {
        const { showForgotPasswordSpiner, forgotPasswordData: { email }, language } = this.props;
        showForgotPasswordSpiner();
        crmForgotPassword.forgotpassword(email.value, language);
    }

    goBack = () => {
        const { navigation } = this.props;
        navigation.goBack();
    }

    render() {
        // StatusBar.setBarStyle('light-content', true);
        const { forgotPasswordData: { email }, setForgotPasswordEmail, forgotPasswordSpiner, navigation, language } = this.props;
        const { isForgotPasswordDisabled } = this.state;
        return (
            <KeyboardAvoidingView style={styles.wrapperAbsolute} behavior={isIOS ? "padding" : null} enabled >
                <StatusBar barStyle="light-content" backgroundColor="#000756"/>
                <ScrollView style={styles.scrollview} keyboardShouldPersistTaps="handled" contentContainerStyle={{ alignItems: 'center', }}>
                    <View style={styles.backgroundBlue}>
                        <Background />
                    </View>
                    <View style={styles.buttonGoBackWrapper}>
                        <ButtonBack id="ForgotPasswordID" navigation={navigation} />
                    </View>
                    <View style={styles.logoWrapper}>
                        <Logo />
                    </View>
                    <View style={styles.container} >
                        <Text style={styles.title}>{localization[language]["Password recovery"]}</Text>
                        <Text style={styles.textDescription}>{localization[language]["Please specify the email you used when registering"]}</Text>
                        <CommonInput
                            keyboardType="email-address"
                            placeholder={localization[language].placeHolder.email}
                            errorText={localization[language].errors.errorEmail}
                            filter={filterEmail}
                            action={setForgotPasswordEmail}
                            isValid={email.isValid}
                            testID="InputEmailForgotPasswordID"
                            textErrorID="EmailErrorRegostrationID"
                            currentLanguageName={language}
                        />
                        <ButtonSubmit
                            text={localization[language].sendBtnText}
                            testID="ButtonSubmitForgotPasswordID"
                            isDisabled={isForgotPasswordDisabled}
                            isShowSpiner={forgotPasswordSpiner}
                            onPress={this.sendForgotPasswordData}
                        />
                    </View>
                </ScrollView>
            </KeyboardAvoidingView>
        )
    }
}

const mapStateToProps = state => ({
    language: state.currentLanguage.language,
    forgotPasswordData: state.forgotPasswordData,
    forgotPasswordSpiner: state.spiners.forgotPasswordSpiner,
    isConnected: state.connectionStatus.isConnected,
});

const mapDispatchToProps = dispatch => bindActionCreators({
    setForgotPasswordEmail,
    showForgotPasswordSpiner
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(ForgotPasswordScreen);