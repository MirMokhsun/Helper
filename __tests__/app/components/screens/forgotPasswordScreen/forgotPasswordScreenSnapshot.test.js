import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ForgotPasswordScreen } from '../../../../../app/components/screens/forgotPasswordScreen/forgotPasswordScreen';

it('create snapshot ForgotPasswordScreen', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<ForgotPasswordScreen
        setForgotPasswordEmail={() => { }}
        showForgotPasswordSpiner={() => { }}
        forgotPasswordData= { {email:"test@com.ua"} }
        language ={"ru"}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});
