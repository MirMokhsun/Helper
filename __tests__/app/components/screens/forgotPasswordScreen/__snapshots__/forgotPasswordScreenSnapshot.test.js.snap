// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot ForgotPasswordScreen 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "#E5E5E5",
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="#000756"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "alignItems": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": -50,
        }
      }
    >
      <Background />
    </View>
    <View
      style={
        Object {
          "height": 25,
          "justifyContent": "flex-start",
          "marginBottom": 10,
          "marginTop": 50,
          "width": "100%",
        }
      }
    >
      <ButtonBack
        id="ForgotPasswordID"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 35,
          "justifyContent": "center",
          "marginVertical": 20,
          "width": "100%",
        }
      }
    >
      <Logo />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "marginVertical": 20,
          "minHeight": 667,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000756",
            "fontFamily": "Roboto",
            "fontSize": 22,
            "fontWeight": "bold",
            "lineHeight": 32,
            "marginVertical": 25,
            "textAlign": "center",
          }
        }
      >
        Восстановление пароля
      </Text>
      <Text
        style={
          Object {
            "color": "#000756",
            "fontFamily": "Roboto-Light",
            "fontSize": 16,
            "lineHeight": 18,
            "marginHorizontal": 20,
            "textAlign": "center",
          }
        }
      >
        Пожалуйста, укажите e-mail, который Вы использовали при регистрации
      </Text>
      <CommonInput
        action={[Function]}
        currentLanguageName="ru"
        errorText="Ошибка ввода e-mail"
        filter={/\\^\\(\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\(\\\\\\.\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\)\\*\\)\\|\\("\\.\\+"\\)\\)@\\(\\(\\\\\\[\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\]\\)\\|\\(\\(\\[a-zA-Z\\\\-0-9\\]\\+\\\\\\.\\)\\+\\[a-zA-Z\\]\\{2,\\}\\)\\)\\$/}
        keyboardType="email-address"
        placeholder="Email"
        testID="InputEmailForgotPasswordID"
        textErrorID="EmailErrorRegostrationID"
      />
      <ButtonSubmit
        isDisabled={true}
        onPress={[Function]}
        testID="ButtonSubmitForgotPasswordID"
        text="Отправить"
      />
    </View>
  </ScrollViewMock>
</KeyboardAvoidingView>
`;
